horace_core/sqw/@DnDBase/parse_args_.m

```diff
@@ -0,0 +1,35 @@
+function args = parse_args_(obj, varargin)
+% Parse the argument passed to the DnD constructor.
+%
+% Return struct with the data set to the appropriate element:
+% - args.filename  % string, presumed to be filename
+% - args.dnd_obj   % DnD class instance
+% - args.sqw_obj   % SQW class instance
+% - args.data_struct % generic struct, presumed to represent DnD
+
+parser = inputParser();
+parser.addOptional('input', [], @(x) (isa(x, 'SQWDnDBase') || is_string(x) || isstruct(x)));
+parser.KeepUnmatched = true;
+parser.parse(varargin{:});
+
+input = parser.Results.input;
+args = struct('dnd_obj', [], 'sqw_obj', [], 'filename', [], 'data_struct', []);
+
+if isa(input, 'SQWDnDBase')
+    if isa(input, class(obj))
+        args.dnd_obj = input;
+    elseif isa(input, 'sqw')
+        args.sqw_obj = input;
+    else
+        error([upper(class(obj)), ':' class(obj)], ...
+            [upper(class(obj)) ' cannot be constructed from an instance of this object "' class(input) '"']);
+    end
+elseif is_string(parser.Results.input)
+    args.filename = input;
```

Comment:
as you have unpacked parser.Results.input as input, may be better to keep the shorter name here

Reply:
Also, `input` is a Matlab built-in, so using a different name would probably be best.